### DISCUSSION TITLE
Moderation enhancements

### DISCUSSION THREAD

**Discussion by brucknerite** [Automattic/wordpress-activitypub#562] [Ideas]
I think it would be better to be able to have ActivityPub comments be better aligned with the general WordPress comments framework:

- If commenting is disabled, ActivityPub comments do not respect that. It is conceivable to be willing to have a blog profile on the Fediverse and not willing comments to appear on the articles themselves.
- There should be a way to automatically approve comment from trusted sources (previous commenters, i.e.)

Are there plans to implement something on these lines? Are these use cases already doable, somehow? What are your thoughts?

Thank you for your attention.

**Comment by Menrath** [comment]
PeerTube introduced a boolean property  `commentsEnabled` for each published object/post/video, which has been adopted by at least by Mobilizon or Lemmy. Though Mobilizon also introduced a own `repliesModerationOption` too which was supposed to be more nuanced, but is not until now. But there is no guarantee that every other software will respect that in the future. Seems what you want is an to give the users an option **to also not be informed about new comments/replies** in the admin user interface if the comments are disabled.

There may be the possibility of introducing a settings that controls this behavior by default (or according to the overall comments settings) and maybe for each post.

Automatically approving comments from trusted sources (ActivityPub actors) should also be technically possible. I haven't researched of how elegant a possible solution could look like though.
